lib/config/config.c

```diff
@@ -394,9 +394,14 @@ config_create(const char *path, cJSON **conf)
 
     *conf = cJSON_ParseWithLength(config, st.st_size + 1);
     if (!*conf) {
-        CLOG_ERR("Failed to parse file as valid JSON (%s)", path);
-        err = merr(EINVAL);
-        goto out;
+        if (cJSON_GetErrorPtr()) {
+            CLOG_ERR("Failed to parse file as valid JSON (%s): %s", path, cJSON_GetErrorPtr());
+            err = merr(EINVAL);
+            goto out;
+        } else {
+            err = merr(ENOMEM);
+            goto out;
+        }
     }
```

Comment:
You could move the "goto out" from line 400 and 403 to line 405. Same comment applies to the other 3 changes as well.

Reply:
Good point. Wasn't really thinking.